test/test_core.py

```diff
@@ -5805,6 +5805,10 @@ def test_fileno(self):
 
   @also_with_noderawfs
   def test_readdir(self):
+    # this test creates new files with specific permissions, and so we must
+    # clear the test dir to get a fresh starting point each time, or else we'd
+    # error on not being able to replace a read-only file in NODERAWFS mode.
+    self.clear()
```

Comment:
I don't think you should need this part of the fix, since `.clear()` should effectively already be called when the test starts.

I think the `remove_readonly` fix below should b enough, no?

Reply:
Unfortunately it is necessary,
```
test_readdir_rawfs (test_core.core0) ... Clearing existing test directory
-- begin program output --
errno: File exists
Aborted(Assertion failed: !err, at: emscripten/test/dirent/test_readdir.c,34,setup)
emscripten/out/test/test_readdir.js:130
      throw ex;
      ^

RuntimeError: Aborted(Assertion failed: !err, at: emscripten/test/dirent/test_readdir.c,34,setup)
```
But you're right that it shouldn't be. I think that might indicate that we still aren't managing to fully clear out the directory...

Do you have a better idea for solving this perhaps? `rm -Rf out/` works as expected. Maybe there's something better than `rmtree()` in python..?